test/memory/MemoryTestPublisher.cpp

```diff
@@ -0,0 +1,393 @@
+
+        // DATA PUBLISHER
+        PublisherAttributes PubDataparam;
+        PubDataparam.topic.topicDataType = "MemoryType";
+        PubDataparam.topic.topicKind = NO_KEY;
+        std::ostringstream pt;
+        pt << "MemoryTest_";
+        if (hostname)
+            pt << asio::ip::host_name() << "_";
+        pt << pid << "_PUB2SUB";
+        PubDataparam.topic.topicName = pt.str();
+        PubDataparam.topic.historyQos.kind = KEEP_ALL_HISTORY_QOS;
+        PubDataparam.topic.historyQos.depth = n_samples;
+        PubDataparam.topic.resourceLimitsQos.max_samples = n_samples + 1;
+        PubDataparam.topic.resourceLimitsQos.allocated_samples = n_samples + 1;
+        PubDataparam.times.heartbeatPeriod.seconds = 0;
+        PubDataparam.times.heartbeatPeriod.fraction = 4294967 * 100;
+        if (!reliable)
+            PubDataparam.qos.m_reliability.kind = BEST_EFFORT_RELIABILITY_QOS;
+        Locator_t loc;
+        loc.port = 15000;
+        PubDataparam.unicastLocatorList.push_back(loc);
+        PubDataparam.properties = property_policy;
+        if (m_data_size > 60000)
+        {
+            PubDataparam.historyMemoryPolicy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+            PubDataparam.qos.m_publishMode.kind = eprosima::fastrtps::ASYNCHRONOUS_PUBLISH_MODE;
+        }
+        mp_datapub = Domain::createPublisher(mp_participant, PubDataparam, (PublisherListener*)&this->m_datapublistener);
+        if (mp_datapub == nullptr)
+            return false;
+
+        //COMMAND PUBLISHER
+        PublisherAttributes PubCommandParam;
+        PubCommandParam.topic.topicDataType = "TestCommandType";
+        PubCommandParam.topic.topicKind = NO_KEY;
+        std::ostringstream pct;
+        pct << "MemoryTest_Command_";
+        if (hostname)
+            pct << asio::ip::host_name() << "_";
+        pct << pid << "_PUB2SUB";
+        PubCommandParam.topic.topicName = pct.str();
+        PubCommandParam.topic.historyQos.kind = KEEP_ALL_HISTORY_QOS;
+        PubCommandParam.topic.historyQos.depth = 100;
+        PubCommandParam.topic.resourceLimitsQos.max_samples = 101;
+        PubCommandParam.topic.resourceLimitsQos.allocated_samples = 101;
+        PubCommandParam.qos.m_durability.kind = TRANSIENT_LOCAL_DURABILITY_QOS;
+        mp_commandpub = Domain::createPublisher(mp_participant, PubCommandParam, &this->m_commandpublistener);
+        if (mp_commandpub == nullptr)
+            return false;
+        SubscriberAttributes SubCommandParam;
+        SubCommandParam.topic.topicDataType = "TestCommandType";
+        SubCommandParam.topic.topicKind = NO_KEY;
+        std::ostringstream sct;
+        sct << "MemoryTest_Command_";
+        if (hostname)
+            sct << asio::ip::host_name() << "_";
+        sct << pid << "_SUB2PUB";
+        SubCommandParam.topic.topicName = sct.str();
+        SubCommandParam.topic.historyQos.kind = KEEP_ALL_HISTORY_QOS;
+        SubCommandParam.topic.historyQos.depth = 100;
```

Comment:
Remove next three lines

Reply:
This was added because of the bug related with the default depth, making the buffer to fill up and freezing the application.